packages/local-cli/core/getCommands.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+'use strict';
+
+const findPlugins = require('./findPlugins');
+const path = require('path');
+const flatten = require('lodash').flatten;
+
+import type { CommandT, ProjectCommandT } from './types.flow';
+
+/**
+ * List of built-in commands
+ */
+const loadLocalCommands = () => [
+  require('../server/server'),
+  require('../runIOS/runIOS'),
+  require('../runAndroid/runAndroid'),
+  require('../library/library'),
+  require('../bundle/bundle'),
+  require('../bundle/ramBundle'),
+  require('../eject/eject'),
+  require('../link/link'),
+  require('../link/unlink'),
+  require('../install/install'),
+  require('../install/uninstall'),
+  require('../upgrade/upgrade'),
+  require('../logAndroid/logAndroid'),
+  require('../logIOS/logIOS'),
+  require('../dependencies/dependencies'),
+  require('../info/info'),
+];
+
+/**
+ * Returns an array of commands that are defined in the project.  
+ * 
+ * This checks all CLI plugins for presence of 3rd party packages that define commands
+ * and loads them
+ */
+const loadProjectCommands = (root: string): Array<ProjectCommandT> => {
```

Comment:
Maybe instead of loading all commands upfront, you could load them as needed? e.g. look one by one, stop looking for more when the command being used is found. Might improve startup time.

Reply:
Good catch. We should improve that in the next PR. Noted.